packages/gensx/tests/secrets.test.tsx

```diff
@@ -0,0 +1,696 @@
+      "123456789012", // definitelyLong - should be masked
+      "456", // nested.short - too short to mask
+      "long-secret-value", // nested.long - should be masked
+    ].join(", ");
+  },
+  { secrets: ["output"] }, // Mark the entire output as containing secrets
+);
+
+const ChildComponent = gsx.Component<
+  { apiKey: string; additionalKey: string },
+  string
+>(
+  "ChildComponent",
+  ({ apiKey, additionalKey }) => `${apiKey}-${additionalKey}`,
+  { secrets: ["apiKey"] },
+);
+
+const ParentComponent = gsx.Component<
+  { credentials: { key: string; other: string } },
+  string
+>(
+  "ParentComponent",
+  ({ credentials }) => (
+    <ChildComponent
+      apiKey={credentials.key}
+      additionalKey={credentials.other}
+      componentOpts={{
+        secrets: ["additionalKey"],
+      }}
+    />
+  ),
+  { secrets: ["credentials"] },
+);
+
+const EdgeCaseComponent = gsx.Component<
+  {
+    config: {
+      emptyString: string;
+      nullValue: null;
+      undefinedValue: undefined;
+      actualSecret: string;
+    };
+  },
+  string
+>(
+  "EdgeCaseComponent",
+  ({ config }) =>
+    JSON.stringify({
+      empty: config.emptyString,
+      secret: config.actualSecret,
+    }),
+  { secrets: ["config"] },
+);
+
+const SpecialCharComponent = gsx.Component<
+  { dotted: string; regex: string; unicode: string; url: string },
+  string
+>("SpecialCharComponent", props => Object.values(props).join(", "), {
+  secrets: ["dotted", "regex", "unicode", "url"],
+});
+
+const ArrayComponent = gsx.Component<{ config: ArrayConfig }, ArrayConfig>(
+  "ArrayComponent",
+  ({ config }) => config,
+  { secrets: ["config"] },
+);
+
+const ReuseComponent = gsx.Component<{}, string[]>(
+  "ReuseComponent",
+  () => {
+    // Return raw values - masking happens at checkpoint level
+    return [
+      "secret.with.dots",
+      "secret.*+?^\\{test\\}",
+      "sēcrēt‿value",
+      "https://api.example.com/secret?key=12345678",
+      "unrelated-string",
+    ];
+  },
+  { secrets: ["output"] }, // Mark the entire output as containing secrets
+);
+
+const PartialComponent = gsx.Component<
+  {
+    prefix: string;
+    suffix: string;
+    middle: string;
+    combined: string;
+    unrelated: string;
+  },
+  string
+>(
+  "PartialComponent",
+  ({ prefix, suffix, middle, combined, unrelated }) =>
+    `${prefix} and ${suffix} and ${middle} and ${combined} and ${unrelated}`,
+  { secrets: ["prefix", "suffix", "middle", "combined"] },
+);
+
+const SecretStreamComponent = gsx.StreamComponent<{ apiKey: string }>(
+  "SecretStreamComponent",
+  ({ apiKey }) => {
+    const stream = function* () {
+      yield "Using API key: ";
+      yield apiKey;
+      yield "\nResponse: ";
+      yield "not sensitive";
+    };
+    return stream();
+  },
+  { secrets: ["apiKey"] }, // Mark both input and output as secret
+);
+
+suite("tree reconstruction", () => {
+  beforeEach(() => {
+    process.env.GENSX_CHECKPOINTS = "true";
```

Comment:
does this need to be reset in the `afterEach`?

Reply:
We don't actually need to set this, so I removed it.